magik-language-server/src/main/java/nl/ramsolutions/sw/magik/languageserver/formatting/FormattingProvider.java

```diff
@@ -42,6 +41,35 @@ public List<TextEdit> provideFormatting(
     return textEdits.stream().map(Lsp4jConversion::textEditToLsp4j).toList();
   }
 
+  /**
+   * Provide formatting for text in range.
+   *
+   * @param magikFile Magik file.
+   * @param options Formatting options.
+   * @param range Range.
+   * @return {@link TextEdit}s.
+   * @throws IOException -
+   */
+  public List<TextEdit> provideRangeFormatting(
+      final MagikFile magikFile, final FormattingOptions options, final Range range) {
+    final List<org.eclipse.lsp4j.TextEdit> textEdits = this.provideFormatting(magikFile, options);
+    return textEdits.stream().filter(edit -> isEditInRange(edit, range)).toList();
+  }
+
+  /**
+   * Test if a given text edit's range intersects with the specified range.
+   *
+   * @param edit the text edit to check
+   * @param range the range to check against
+   * @return true if the text edit's range intersects with the specified range, false otherwise
+   */
+  private boolean isEditInRange(TextEdit edit, Range range) {
+    Position editStart = Lsp4jConversion.positionFromLsp4j(edit.getRange().getStart());
```

Comment:
final

Reply:
And try to reduce the nesting of expressions. As a rule of thumb I like to prevent method calls within method calls (or rather, arguments).